### ISSUE TITLE
Microsoft.Identity.Client version 1.1.2-preview0008 doesn't work properly in Xamarin

### ISSUE THREAD

**Issue by jpdc-developer** [AzureAD/microsoft-authentication-library-for-dotnet#532]
**Edit:**  I've followed [this wiki page](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/Troubleshooting-Xamarin.Android-issues-with-MSAL) so that my nuget packages are up to date. I am still observing the same issue.

I used Microsoft.Identity.Client version 1.0.303282006 for authentication in my Xamarin App but I have updated it to version 1.1.2-preview0008.

It behaves incorrectly now. After refactoring the code to work with version 1.1.2-preview0008, the app builds, but when the line of code for getting the token is run, `AcquireTokenAsync(Scopes)`, this exception is raised:

    "Method 'Microsoft.Identity.Client.PublicClientApplication.AcquireTokenAsync' not found."

I tried other signatures of the same method. Right now, I’m using `_pca.AcquireTokenAsync(Scopes, uIParent)`. I can get uIParent from the Android App, but it doesn’t work 100%. It opens a login dialog, allows me to enter my username, then it shows a page saying “We’re unable to complete your request.”

I had a look at the url of this page and it indicates that there’s been an error:

> https://login.live.com/err.srf?lc=2057#error=invalid_request&error_description=The+provided+value+for+the+input+parameter+'redirect_uri'+is+not+valid.+The+expected+value+is+'https://login.live.com/oauth20_desktop.srf'+or+a+URL+which+matches+the+redirect+URI+registered+for+this+client+application

I’ve tried other similar signatures of the same method and it always fails in the part where it’s supposed to ask me for my password. It seems like my Microsoft.Identity.Client nuget package can be found but it’s not being utilised properly.

From the looks of things, it seems like the preview version (1.1.2-preview0008) of Microsoft.Identity.Client isn’t working properly on the Xamarin App. However, I tried making a WinForms app and it works properly there so I'm not sure what caused this issue.

**Comment by jennyf19** [comment]
@jpdc-developer Since 1.1.0-preview, [AcquireTokenAsync(Scopes)](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/dev/tests/dev%20apps/XForms/XForms/App.xaml.cs#L45) has been unavailable to Android applications. Please use one of these [methods](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/dev/src/Microsoft.Identity.Client/PublicClientApplication.cs#L204) instead.

Based on the error message, looks like the redirectUri is incorrect. Please check out our [Xamarin Forms sample app](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/dev/tests/dev%20apps/XForms/XForms/App.xaml.cs#L45) for more details.

**Comment by jmprieur** [comment]
@jpdc-developer.  Did you clean build? It seems to me that there is still a mixture of versions (Nuget mismatched?)
then as Jenny says the redirectUrl needs to match your application ID.
``msal<clientId>://auth`` 
For instance see the following sample: https://github.com/Azure-Samples/active-directory-xamarin-native-v2/blob/master/UserDetailsClient/UserDetailsClient.Droid/MainActivity.cs#L23

**Comment by jpdc-developer** [comment]
You're right @jennyf19. I needed to set the redirect URI and use` AcquireTokenAsync(IEnumerable<string> scopes, UIParent parent)`. I also followed the way the Xamarin Forms Sample App set things up so that I had to add more setup code in the Android Manifest. 

I also added a custom redirect URI in the[ Application Registration Portal](https://apps.dev.microsoft.com/) in the format `msal<clientId>://auth`.

**Comment by jennyf19** [comment]
Thanks for the follow-up, @jpdc-developer - glad everything is working for you now.
